sans/direct-beam.ipynb

```diff
@@ -0,0 +1,529 @@
+    "\n",
+    "path = dataconfig.data_root\n",
+    "direct_beam_file = 'DirectBeam_20feb_full_v3.dat' # same as DirectBeam_28Apr2020.dat\n",
+    "moderator_file = 'ModeratorStdDev_TS2_SANS_LETexptl_07Aug2015.txt'\n",
+    "sample_run_number = 49338\n",
+    "sample_transmission_run_number = 49339\n",
+    "background_run_number = 49334\n",
+    "background_transmission_run_number = 49335\n",
+    "\n",
+    "def load_larmor(run_number):\n",
+    "    return sc.neutron.load(filename=f'{path}/LARMOR000{run_number}.nxs')\n",
+    "\n",
+    "def load_rkh(filename):\n",
+    "    return sc.neutron.load(\n",
+    "           filename=filename,\n",
+    "           mantid_alg='LoadRKH',\n",
+    "           mantid_args={'FirstColumnValue':'Wavelength'})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "sample_trans = load_larmor(sample_transmission_run_number)\n",
+    "sample = load_larmor(sample_run_number)\n",
+    "background_trans = load_larmor(background_transmission_run_number)\n",
+    "background = load_larmor(background_run_number)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizations and mask setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "masks = {}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Bad straws"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from scipp.plot import plot\n",
+    "from loki import LoKI\n",
+    "loki = LoKI()\n",
+    "spectrum_counts = sc.sum(sample, 'tof') # sum is optional, could also keep TOF\n",
+    "pixel_counts = loki.to_logical_dims(spectrum_counts)\n",
+    "straw_counts = sc.sum(pixel_counts, 'pixel')\n",
+    "plot(straw_counts, log=True, vmin=1, vmax=7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bad_straws = sc.Variable(dims=['spectrum'], shape=sample.coords['spectrum'].shape, dtype=sc.dtype.bool)\n",
+    "mask = sc.Variable(value=True)\n",
+    "bad_straws[loki.straw(tube=0, straw=6)] |= mask\n",
+    "bad_straws[loki.straw(tube=1, straw=2)] |= mask\n",
+    "bad_straws[loki.straw(tube=5, straw=1)] |= mask\n",
+    "for straw in 1,2,3,4,5:\n",
+    "    bad_straws[loki.straw(tube=12, straw=straw)] |= mask\n",
+    "masks['bad-straws'] = bad_straws"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Bad pixels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plot(pixel_counts, vmax=100, axes=['straw', 'tube', 'pixel'])"
```

Comment:
Slightly confusing that the plots here are not using the masks set up in previous steps. Maybe rearrange to move mask generation closer to point of use?

Reply:
The main purpose of this was showing some tools that could be used during, e.g., commissioning. I am not adding the masks to the data array because actually we later need it for another data arrays as well (sample and background). I thought adding the masks to both would have distracted from the point.